metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_api.py

```diff
@@ -234,42 +240,45 @@ def authenticate(self, connection_args: "DremioSourceConfig") -> None:
             "Credentials cannot be refreshed. Please check your username and password."
         )
 
-    def get(self, url: str) -> Dict:
-        """execute a get request on dremio"""
-        logger.debug(f"GET request to {self.base_url + url}")
-        self.report.api_calls_total += 1
-        self.report.api_calls_by_method_and_path["GET " + url] += 1
-
-        with PerfTimer() as timer:
-            response = self.session.get(
-                url=(self.base_url + url),
-                verify=self._verify,
-                timeout=self._timeout,
-            )
-            self.report.api_call_secs_by_method_and_path["GET " + url] += (
-                timer.elapsed_seconds()
-            )
-            # response.raise_for_status()  # Enabling this line, makes integration tests to fail
-            return response.json()
+    def _request(self, method: str, url: str, data: str | None = None) -> Dict:
+        """Send a request to the Dremio API."""
 
-    def post(self, url: str, data: str) -> Dict:
-        """execute a get request on dremio"""
-        logger.debug(f"POST request to {self.base_url + url}")
+        logger.debug(f"{method} request to {self.base_url + url}")
         self.report.api_calls_total += 1
-        self.report.api_calls_by_method_and_path["POST " + url] += 1
+        self.report.api_calls_by_method_and_path[f"{method} {url}"] += 1
 
         with PerfTimer() as timer:
-            response = self.session.post(
+            response = self.session.request(
+                method=method,
                 url=(self.base_url + url),
                 data=data,
                 verify=self._verify,
                 timeout=self._timeout,
             )
-            self.report.api_call_secs_by_method_and_path["POST " + url] += (
+            self.report.api_call_secs_by_method_and_path[f"{method} {url}"] += (
                 timer.elapsed_seconds()
             )
             # response.raise_for_status()  # Enabling this line, makes integration tests to fail
-            return response.json()
+            try:
+                return response.json()
+            except requests.exceptions.JSONDecodeError as e:
+                logger.info(
```

Comment:
```suggestion
                logger.error(
```

Reply:
at this point it might just be an info; the exception being raised ensures that the caller is responsible for handling + logging it as an error if appropriate